src/ignore_gradients.jl

```diff
@@ -0,0 +1,50 @@
+"""
+    ignore_gradients(f::Function)
+
+Tells the AD system to ignore the gradients of the wrapped closure. The primal computation
+(forward pass) is executed normally.
+
+```julia
+ignore_gradients() do
+    value = rand()
+    push!(collection, value)
+end
+```
+
+Using this incorrectly could lead to incorrect gradients. E.g.
+```julia
+function wrong_grads(x)
+    y = ones(3)
+    ignore_gradients() do
+        push!(y, x)
+    end
+    return sum(y)
+end
+```
+"""
+ignore_gradients(f::Function) = f()
+
+"""
+    ignore_gradients(x)
```

Comment:
might it be worthwhile to explicitly comment on behaviour for functors ?

Reply:
added to the docs, but in the docstring it should be clear from the dispatch on `Function`